app/models/user.rb

```diff
@@ -9,8 +9,22 @@ class User < ActiveRecord::Base
 										format: { with: /\A[a-z0-9._+-]+@[a-z0-9.-]+\.[a-z]+\z/ }
 
 	before_validation :downcase_email
+	before_create :generate_activation_token
 
-	def downcase_email 
+	def downcase_email
 		self.email = email.downcase if self.email
 	end
-end
\ No newline at end of file
+
+  def activate!
+    self.update_attribute('activation_token', nil)
+  end
+
+	private
+
+	def generate_activation_token
+		if self.activation_token.blank?
+			self.activation_token = "#{SecureRandom.hex(8)}"
```

Comment:
Oui, it would be sufficient. Interpolated string bo w pewnym momencie tworzyłam token z SecureRandom + email :P już zmieniam